src/ck-ldb/CentralPredictor.C

```diff
@@ -202,26 +202,20 @@ void CentralLB::FuturePredictor(BaseLB::LDStats* stats) {
 
   if (predicted_model->cur_stats < _lb_predict_delay) {
     // not yet ready to create the model, just store the relevant statistic
-    predicted_model->collection[predicted_model->start_stats].objData.resize(stats->n_objs);
-    predicted_model->collection[predicted_model->start_stats].commData.resize(stats->n_comm);
-    predicted_model->collection[predicted_model->start_stats].n_objs = stats->n_objs;
     predicted_model->collection[predicted_model->start_stats].n_migrateobjs = stats->n_migrateobjs;
-    predicted_model->collection[predicted_model->start_stats].n_comm = stats->n_comm;
-    for (i=0; i<stats->n_objs; ++i)
-      predicted_model->collection[predicted_model->start_stats].objData[i] = stats->objData[i];
-    for (i=0; i<stats->n_comm; ++i)
-      predicted_model->collection[predicted_model->start_stats].commData[i] = stats->commData[i];
+    predicted_model->collection[predicted_model->start_stats].objData = stats->objData;
+    predicted_model->collection[predicted_model->start_stats].commData = stats->commData;
     ++predicted_model->cur_stats;
     ++predicted_model->start_stats;
 
   } else {
-
+    const int n_objs = stats->objData.size();
     if (predicted_model->parameters == NULL) {     // time to create the new prediction model
       // allocate parameters
-      predicted_model->model_valid = new bool[stats->n_objs];
-      predicted_model->parameters = new double*[stats->n_objs];
-      for (i=0; i<stats->n_objs; ++i) predicted_model->parameters[i] = new double[predicted_model->predictor->num_params];
-      for (i=0; i<stats->n_objs; ++i) {
+      predicted_model->model_valid = new bool[n_objs];
```

Comment:
Would be nicer to have these as vectors as well

Reply:
Done.,